src/channel.rs

```diff
@@ -1,73 +1,1022 @@
+    ///
+    /// ```
+    /// use rss::{ChannelBuilder, Channel};
+    /// use rss::extension::itunes::{ITunesChannelExtensionBuilder,
+    /// ITunesOwnerBuilder, ITunesCategoryBuilder};
+    ///
+    /// let owner = ITunesOwnerBuilder::new()
+    ///     .email(Some("email@example.com".to_owned()))
+    ///     .name(Some("name".to_owned()))
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let subcategory = ITunesCategoryBuilder::new()
+    ///     .text("text")
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let category = ITunesCategoryBuilder::new()
+    ///     .text("text")
+    ///     .subcategory(Some(Box::new(subcategory)))
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let categories = vec![category];
+    ///
+    /// let itunes_channel = ITunesChannelExtensionBuilder::new()
+    ///     .author(Some("author".to_owned()))
+    ///     .block(Some("block".to_owned()))
+    ///     .image(Some("image".to_owned()))
+    ///     .explicit(Some("explicit".to_owned()))
+    ///     .subtitle(Some("subtitle".to_owned()))
+    ///     .summary(Some("summary".to_owned()))
+    ///     .keywords(Some("keywords".to_owned()))
+    ///     .new_feed_url(Some("new_feed_url".to_owned()))
+    ///     .complete(Some("complete".to_owned()))
+    ///     .owner(Some(owner))
+    ///     .categories(categories)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let channel = ChannelBuilder::new()
+    ///     .itunes_ext(Some(itunes_channel))
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// assert!(channel.itunes_ext().is_some());
+    /// ```
+    ///
+    /// ```
+    /// use rss::{ChannelBuilder, Channel};
+    ///
+    /// let channel = ChannelBuilder::new()
+    ///     .itunes_ext(None)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// assert!(channel.itunes_ext().is_none());
+    /// ```
+    pub fn itunes_ext(&self) -> Option<ITunesChannelExtension>
+    {
+        self.itunes_ext
+            .clone()
+    }
+
+    /// TODO: document dublincore getter
+    pub fn dublin_core_ext(&self) -> Option<DublinCoreExtension>
+    {
+        self.dublin_core_ext
+            .clone()
+    }
+
+    /// TODO: document extensions getter
+    pub fn extensions(&self) -> ExtensionMap
+    {
+        self.extensions
+            .clone()
+    }
+
+    /// TODO: document namespaces getter
+    pub fn namespaces(&self) -> HashMap<String, String>
+    {
+        self.namespaces
+            .clone()
+    }
 }
 
-impl Channel {
+impl Channel
+{
+    /// Construct a `Channel` from a url string.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// extern crate rss;
+    ///
+    /// use rss::Channel;
+    ///
+    /// fn main()
+    /// {
+    ///     let url = "https://feedpress.me/usererror.xml";
+    ///
+    ///     Channel::from_url(url).unwrap();
+    /// }
+    /// ```
+    pub fn from_url(url: &str) -> Result<Channel, Error>
```

Comment:
I'm wondering if this function should be gated behind a feature flag in case the user wants to use a different networking library. That way we wouldn't be including an unneeded dependency.